olp-cpp-sdk-core/src/client/OlpClient.cpp

```diff
@@ -240,31 +240,44 @@ std::shared_ptr<http::NetworkRequest> OlpClient::CreateRequest(
 }
 
 NetworkAsyncCallback GetRetryCallback(
-    int current_try, int current_backdown_period, const RetrySettings& settings,
+    int current_try, int current_backdown_period,
+    long long accumulated_wait_time, const RetrySettings& settings,
     const NetworkAsyncCallback& callback,
     const std::shared_ptr<http::NetworkRequest>& network_request,
     std::weak_ptr<http::Network> network,
     const std::weak_ptr<CancellationContext>& weak_cancel_context) {
   ++current_try;
   return [=](HttpResponse response) {
+    const long long max_wait_time =
+        settings.timeout * 1000ll;  // convert seconds to msecs
```

Comment:
```suggestion
        settings.timeout * 1000ll;
```
The comment is not needed if an abstract type is used.